runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/BpmnModeler/RouteLeavingGuard.tsx

```diff
@@ -0,0 +1,62 @@
+import { Location } from 'history';
+import React, { useEffect, useState } from 'react';
+import { Prompt } from 'react-router-dom';
+import { translate } from 'src/hooks/useTranslations';
+import LeavePrompt from './LeavePrompt';
+
+interface Props {
+  when?: boolean | undefined;
+  navigate: (path: string) => void;
+  shouldBlockNavigation: (location: Location) => boolean;
+}
+
+const RouteLeavingGuard = ({
+  when,
+  navigate,
+  shouldBlockNavigation,
+}: Props) => {
+  const [modalVisible, setModalVisible] = useState(false);
+  const [lastLocation, setLastLocation] = useState<Location | null>(null);
+  const [confirmedNavigation, setConfirmedNavigation] = useState(false);
```

Comment:
it should be more like `isNavigationConfirmed / setNavigationConfirmed` because it sounds like it has a string in it

Reply:
done